python/lsst/daf/persistence/policy.py

```diff
@@ -69,25 +69,36 @@ class Policy(_PolicyBase):
     - pex policy: read is supported, although this is deprecated and will at some point be removed.
     """
 
-    def __init__(self, other=None, preference=('data', 'policy', 'pexPolicy', 'filePath', 'defaultInitData'),
-                 **kwargs):
-        """Constructor. Accepts an optional vars, one of which will be used to initalize the Policy.
-        If more than initializer var is populated, one will be selected in order specified by 'preference'.
-
-        :param preference: a tuple in order of preference to be used if more than 1 initialzer var is
-                           populated.
-        :param kwargs:
-            data: a dict with initial key-value pairs.
-            policy: an instance of a Policy that should be copied into this Policy.
-            filePath: path to a policy file. May be .yaml or .paf
-            pexPolicy: an instance of a pex policy to copy into this Policy
-            defaultInitData: policy data with a populated instance of a defaultInitData
-                             e.g. Policy.defaultInitData('daf_persistence', 'testPolicy.yaml', 'tests')
-
-        :return:
+    def __init__(self, other=None):
+        """Initialize the Policy. Other can be used to initialize the Policy in a variety of ways:
+        other (string) Treated as a path to a policy file on disk. Must end with '.paf' or '.yaml'.
+        other (Pex Policy) Initializes this Policy with the values in the passed-in Pex Policy.
+        other (Policy) Copies the other Policy's values into this one.
+        other (dict) Copies the values from the dict into this Policy.
+    
         """
         collections.UserDict.__init__(self)
-        self.__init(other, preference, **kwargs)
+        
+        if other is None:
+            return
+
+        try:
+            other.keys() # is it a dict?
```

Comment:
Also, why not just ask if it's dict-like and fit it in with the rest of the tests below?